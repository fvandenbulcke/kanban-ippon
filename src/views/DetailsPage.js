import { useParams } from "react-router-dom";

import './DetailsPage.scss';

const task = {
  id: 1,
  title: 'my super feature',
  status: 'DOING',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce suscipit.'
}

function DetailsPage(){
  const { id } = useParams();

  return (
    <div>
      <h1 className="title">Kanban: Gestion de projet</h1>
      <div className="task-details">
        <div>
          <div className="task-details-section-title">Details: </div>
          <div className="task-details-section-content">
            <div data-testid="task-id">
              <span className="task-details-section-content-title">task: </span>
              <span>{id}</span>
            </div>
            <div data-testid="task-title">
              <span className="task-details-section-content-title">Feature: </span>
              <span>{task.title}</span>
            </div>
            <div data-testid="task-status">
              <span className="task-details-section-content-title">Status: </span>
              <span>{task.status}</span>
            </div>
          </div>
        </div>
        <div data-testid="task-description">
          <div className="task-details-section-title">Description: </div>
          <div>{task.description}</div>
        </div>
      </div>
    </div>

  )
}

export default DetailsPage;
