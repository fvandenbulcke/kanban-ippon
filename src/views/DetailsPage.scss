.task-details {
  padding: 20px;
  border: black solid 2px;
  width: 50%;
  margin: auto;

  &-section {

    &-title {
      font-size: x-large;
      text-decoration: underline;
      font-weight: bold;
      margin: 20px 0px 10px;
    }

    &-content {
      display: grid;
      grid-template-columns: repeat(3,1fr);

      &-title {
        font-weight: bold;
        margin-right: 10px;
      }
    }
  }
}