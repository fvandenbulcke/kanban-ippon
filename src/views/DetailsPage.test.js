import { render, screen } from '@testing-library/react';
import DetailsPage from "./DetailsPage";

jest.mock('react-router-dom', () => ({
  useParams: () => ({ id: 1 })
}))

describe('<DetailsPage/>', () => {
  test('should render', () => {
    const {container} = render(<DetailsPage />);
    expect(container).toMatchSnapshot();
  });

  test('should display task details', () => {
    render(<DetailsPage />);
    const taskId = screen.getByTestId('task-id');
    expect(taskId).toHaveTextContent('task: 1');
    const taskTitle = screen.getByTestId('task-title');
    expect(taskTitle).toHaveTextContent('Feature: my super feature');
    const taskStatus = screen.getByTestId('task-status');
    expect(taskStatus).toHaveTextContent('Status: DOING');
    const taskDescription = screen.getByTestId('task-description');
    expect(taskDescription).toHaveTextContent('Description: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce suscipit.');
  });
})