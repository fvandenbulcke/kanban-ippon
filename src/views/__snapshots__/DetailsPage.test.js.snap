// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsPage/> should render 1`] = `
<div>
  <div>
    <h1
      class="title"
    >
      Kanban: Gestion de projet
    </h1>
    <div
      class="task-details"
    >
      <div>
        <div
          class="task-details-section-title"
        >
          Details: 
        </div>
        <div
          class="task-details-section-content"
        >
          <div
            data-testid="task-id"
          >
            <span
              class="task-details-section-content-title"
            >
              task: 
            </span>
            <span>
              1
            </span>
          </div>
          <div
            data-testid="task-title"
          >
            <span
              class="task-details-section-content-title"
            >
              Feature: 
            </span>
            <span>
              my super feature
            </span>
          </div>
          <div
            data-testid="task-status"
          >
            <span
              class="task-details-section-content-title"
            >
              Status: 
            </span>
            <span>
              DOING
            </span>
          </div>
        </div>
      </div>
      <div
        data-testid="task-description"
      >
        <div
          class="task-details-section-title"
        >
          Description: 
        </div>
        <div>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce suscipit.
        </div>
      </div>
    </div>
  </div>
</div>
`;
